packages/gatsby/src/utils/find-page-by-path.ts

```diff
@@ -46,7 +46,7 @@ export function findPageByPath(
   }
 
   // we didn't find exact static page, time to check matchPaths
-  for (const [, page] of pages) {
+  for (const page of pages.values()) {
```

Comment:
This is just fixing lookup for client only paths in engine. In general it's quite not ideal to hit this code path as it cause cause severe memory spikes for large sites and ideally we have something nicer to handle those paths than iterating every page later on. (/cc @vladar about issue we talked about some time ago)

Reply:
we probably want to move separate list of matchPaths only